crates/router/src/connector/stripe.rs

```diff
@@ -1759,11 +1767,31 @@ impl api::IncomingWebhook for Stripe {
                 )
             }
             stripe::WebhookEventObjectType::Refund => {
-                api_models::webhooks::ObjectReferenceId::RefundId(
-                    api_models::webhooks::RefundIdType::ConnectorRefundId(
-                        details.event_data.event_object.id,
+                match details.event_data.event_object.metadata {
+                    // if meta_data is present
+                    Some(meta_data) => {
+                        // Issue: 2076
+                        match meta_data.is_refund_id_as_reference {
+                            // if the order_id is refund_id
+                            Some(true) => api_models::webhooks::ObjectReferenceId::RefundId(
+                                api_models::webhooks::RefundIdType::RefundId(meta_data.order_id),
```

Comment:
In Refund webhooks, do we get metadata of refund request / payment request? Were you able to test this?

Reply:
Yes, refunds tested too